### PR TITLE
Skip Pull Request Checks / Verify Description Checklist for dependabot

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Skips the CI check `Pull Request Checks / Verify Description Checklist` for PRs from dependabot.

See the CI checks on a dependabot PR like [this](https://github.com/opensearch-project/OpenSearch/pull/13117) where the check fails since it doesn't contain the checklist on the PR template.

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ashking94 — APPROVED** [review]
Thanks for this change. LGTM.

**Comment by cwperks** [comment]
> Tests with failures:
 - org.opensearch.cluster.coordination.AwarenessAttributeDecommissionIT.testConcurrentDecommissionAction

Impacted by flaky test: https://github.com/opensearch-project/OpenSearch/issues/12197

**Comment by cwperks** [comment]
> Tests with failures:
 - org.opensearch.ratelimitting.admissioncontrol.AdmissionForClusterManagerIT.testAdmissionControlResponseStatus

Impacted by https://github.com/opensearch-project/OpenSearch/issues/13045

File: .github/workflows/pull-request-checks.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: peternied/check-pull-request-description-checklist@v1.1
+        if: github.actor != 'dependabot[bot]'
```
Comment (peternied) [inline .github/workflows/pull-request-checks.yml:21]: Looks like this was inside `${{` `}}` previously, have you been able to test that this works in your repo?
https://github.com/opensearch-project/OpenSearch/blob/bad49efab29bd5161458f182b87c5d78d8de3cac/.github/workflows/dependabot_pr.yml#L10

Comment (cwperks) [reply .github/workflows/pull-request-checks.yml:21]: @peternied Yes, I tested this on my fork by pushing the change to main and re-basing one of the existing [dependabot PRs](https://github.com/cwperks/OpenSearch/pull/149). Should I add in the curly braces?

**Review by kotwanikunal — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Comment by peternied** [comment]
@cwperks Could you see about backporting this change into the 2.x branch?

**Comment by cwperks** [comment]
@peternied 2.x does not have the pull-request-check workflow. Typically PRs to 2.x will not include the checklist so does it make sense to backport the workflow?

**Comment by peternied** [comment]
> @peternied 2.x does not have the pull-request-check workflow. Typically PRs to 2.x will not include the checklist so does it make sense to backport the workflow?

Actually that sounds like another good criteria - in the 2.x line - but if features are directly going into 2.x still seems like they should use the same template.  What do you think about opening up an issue that could be discussed during triage?

**Pull request merged**